ksqldb-engine/src/main/java/io/confluent/ksql/properties/PropertyOverrider.java

```diff
@@ -21,11 +21,26 @@
 import io.confluent.ksql.parser.tree.SetProperty;
 import io.confluent.ksql.parser.tree.UnsetProperty;
 import io.confluent.ksql.statement.ConfiguredStatement;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
 import io.confluent.ksql.util.KsqlStatementException;
+import java.util.Arrays;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
 
 public final class PropertyOverrider {
 
+  private static final Set<String> QUERY_LEVEL_CONFIGS = new HashSet<>(Arrays.asList(
+      "auto.offset.reset",
+      "buffered.records.per.partition",
+      "cache.max.bytes.buffering",
+      "default.deserialization.exception.handler",
+      "default.timestamp.extractor",
+      "max.task.idle.ms",
+      "task.timeout.ms",
+      "topology.optimization"));
```

Comment:
Do we not already have a list of query-level configs maintained somewhere within the ksql repo? Or is that only hard-coded into the front-end?

IIRC from @tolgadur's work on the new `list properties` fields, we do have something like this that explicitly marks the query-level configs -- we should make sure to reuse that, or consolidate the two in some other way, so we don't have to worry about maintaining  multiple copies and keeping them all up to date

Reply:
ah I didn't see that I will reuse it instead